### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by tkhurana96**

### CODE REVIEW

File: cpp-straight-calls-withErrorHandling/tcp-client.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+std::string getErrorMsg()
+{
+	std::mutex lockToErrorString;
```
Comment (agauniyal) [inline cpp-straight-calls-withErrorHandling/tcp-client.cpp]: should be static..

File: cpp-straight-calls-withErrorHandling/tcp-server.cpp [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include <unistd.h>  // close()
+
+std::string getErrorMsg()
+{
+	std::mutex lockToErrorString;
```
Comment (agauniyal) [inline cpp-straight-calls-withErrorHandling/tcp-server.cpp]: again, should be static?

**Comment by tkhurana96** [comment]
made getErrorMsg() static

File: cpp-straight-calls-withErrorHandling/tcp-client.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+std::string getErrorMsg()
+{
+	static std::mutex lockToErrorString;
+	lockToErrorString.lock();
+	char *errMsg = strerror(errno);
+	std::string returnString(errMsg);
```
Comment (agauniyal) [inline cpp-straight-calls-withErrorHandling/tcp-client.cpp]: what if exception occurs while creation of this string?

**Review by agauniyal — CHANGES_REQUESTED** [review]

**Comment by tkhurana96** [comment]
Added lockguard to getErrorMsg()

**Review by agauniyal — APPROVED** [review]

**Pull request merged**